### PR TITLE
[BugFix] Fix checkReplicateNormal error mistake drop table bug

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/statistic/StatisticExecutor.java [hunk 1]
```diff
@@ -219,9 +219,11 @@ public void collectStatistics(StatisticsCollectJob tcj) {
 
             GlobalStateMgr.getCurrentStatisticStorage().expireColumnStatistics(table, columns);
         } catch (Exception e) {
-            analyzeJob.setWorkTime(LocalDateTime.now());
-            analyzeJob.setReason(e.getMessage());
-            GlobalStateMgr.getCurrentAnalyzeMgr().updateAnalyzeJobWithLog(analyzeJob);
+            if (analyzeJob.getId() != -1) {
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/statistic/StatisticExecutor.java:222]: Please add comments to explain why -1 is exception.

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeChecker.java [hunk 2]
```diff
@@ -417,6 +417,11 @@ public Void visitAnalyzeStatement(AnalyzeStmt statement, ConnectContext session)
         @Override
         public Void visitCreateAnalyzeJobStatement(CreateAnalyzeJobStmt statement, ConnectContext session) {
             TableName tableName = statement.getTableName();
+            if (tableName == null) {
+                //CREATE ANALYZE FULL ALL
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeChecker.java:421]: nit
```suggestion
                // CREATE ANALYZE FULL ALL
```

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 25 / 57 (43.86%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/SampleStatisticsCollectJob.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|0|4|00.00%|
|:large_blue_circle:|com/starrocks/statistic/StatisticsMetaManager.java|0|14|00.00%|
|:large_blue_circle:|com/starrocks/statistic/TableCollectJob.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|25|37|67.57%|

File: fe/fe-core/src/main/java/com/starrocks/statistic/StatisticExecutor.java (hunk 1 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/statistic/StatisticExecutor.java:225]: Add comment done.

// If the job id is equal to -1, it represents the automatic full statistics collection of the new version.
// Automatic full statistics collection is an implicit task
// that is not recorded in the analyze job list, nor does it update the status of the analyze job.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 25 / 57 (43.86%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/SampleStatisticsCollectJob.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|0|4|00.00%|
|:large_blue_circle:|com/starrocks/statistic/StatisticsMetaManager.java|0|14|00.00%|
|:large_blue_circle:|com/starrocks/statistic/TableCollectJob.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|25|37|67.57%|

File: fe/fe-core/src/main/java/com/starrocks/statistic/StatisticExecutor.java [hunk 3]
```diff
@@ -219,9 +219,14 @@ public void collectStatistics(StatisticsCollectJob tcj) {
...
             GlobalStateMgr.getCurrentStatisticStorage().expireColumnStatistics(table, columns);
         } catch (Exception e) {
-            analyzeJob.setWorkTime(LocalDateTime.now());
-            analyzeJob.setReason(e.getMessage());
-            GlobalStateMgr.getCurrentAnalyzeMgr().updateAnalyzeJobWithLog(analyzeJob);
+            // If the job id is equal to -1, it represents the automatic full statistics collection of the new version.
+            // Automatic full statistics collection is an implicit task
+            // that is not recorded in the analyze job list, nor does it update the status of the analyze job.
+            if (analyzeJob.getId() != -1) {
+                analyzeJob.setStatus(Constants.ScheduleStatus.FAILED);
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/statistic/StatisticExecutor.java:226]: There are no need to setStatus and setReason

**Review by Youngwb — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Pull request merged**